src/component.js

```diff
@@ -38,9 +38,9 @@ export function Component(props, context) {
  * updated
  */
 Component.prototype.setState = function(update, callback) {
+	if (this.state) this._prevState = this.state;
```

Comment:
A thought:  could we (ab)use a getter/setter pair on `LegacyComponent#state` to restore the old behaviour?

```js
// emulates forward-looking .state by storing values in ._state and prioritizing return of `._nextState`.
const stateDescriptor = {
  get() {
    return this._nextState || this._state;
  },
  set(newState) {
    this._state = newState;
  }
};

function LegacyComponent(p, c) {
  Object.defineProperty(this, 'state', stateDescriptor);
  Component.call(this, p, c);
}
```